methods/incidentReports.coffee

```diff
@@ -40,9 +41,26 @@ Meteor.methods
     user = Meteor.user()
     if not Roles.userIsInRole(user._id, ['admin'])
       throw new Meteor.Error("auth", "User does not have permission to edit incidents")
-    incident.modifiedByUserId = user._id
-    incident.modifiedByUserName = user.profile.name
-    res = Incidents.update(incident._id, incident)
+
+    # Remove existing type props if user changes incident type and merge incident
+    # from client with existing incident
+    if incident.cases
+      fieldsToRemove = deaths: true, specify: true
+    else if incident.deaths
+      fieldsToRemove = cases: true, specify: true
+    else if incident.specify
+      fieldsToRemove = cases: true, deaths: true
+    existingIncident = Incidents.findOne(incident._id)
+    updatedIncident = _.extend(existingIncident, incident)
```

Comment:
Keeping the properties from the old incident makes this method almost the same as updateIncidentReport. We should combine them into a single method.